### ISSUE TITLE
bug: Component Stylesheet Loading Issue with Multiple `styleUrls`

### ISSUE THREAD

**Issue by a7medm7med** [ionic-team/stencil#5251]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.9.1

### Current Behavior

When using multiple stylesheets (`styleUrls`) for a Stencil.js component, modifying one of the stylesheet files causes the other stylesheet to be removed from the rendered element. This behavior is unexpected and leads to inconsistent styling when developers attempt to update or modify stylesheets independently.

### Expected Behavior

Modifying one stylesheet file should not interfere with the loading and application of other stylesheets defined in the `styleUrls` array. Each stylesheet should be independently loaded and applied to the component.

### System Info

```shell
System: node 18.19.0
    Platform: windows (10.0.22631)
   CPU Model: 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz (12 cpus)
    Compiler: D:\Projects\web-components\node_modules\.pnpm\@stencil+core@4.9.1\node_modules\@stencil\core\compiler\stencil.js
       Build: 1704730339
     Stencil: 4.9.1
  TypeScript: 5.2.2
      Rollup: 2.42.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.26.0
```

### Steps to Reproduce

1. Create component with multiple `styleUrls`, such as:
    ```tsx
    @Component({
      tag: 'custom-button',
      styleUrls: ['button.background.scss', 'button.text.scss'],
      shadow: true,
    })
    export class Button {
      render() {
        return <button class="button">Button Text</button>;
      }
    }
    ```
2. Add distinct styles to each stylesheet file (e.g., `button.background.scss` and `button.text.scss`).
3. Start the Stencil.js dev server using the `npm run start` command.
4. Observe the rendered component with both styles applied correctly.
5. Modify one of the stylesheet files (e.g., `button.text.scss`).
6. Observe that the styles from the other stylesheet (e.g., `button.background.scss`) are no longer applied to the rendered component.

### Code Reproduction URL

https://github.com/a7medm7med/stencil-bug

### Additional Information

I believe this frustrating issue is also connected to the equally vexing problem in issue #2635.

**Comment by christian-bromann** [comment]
Thanks! I appreciate the time to put together a minimal reproduction case. I've confirmed that this is a bug with the reproduction case. However, I'm going to close this as a duplicate of #2635. Thanks again!

**Comment by christian-bromann** [comment]
Actually this issue is not a duplicate of #2635, so I will re-open. I however have a fix for this in place in #5244

**Comment by christian-bromann** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.11.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.11.0).
